File: 7KYU/Growth ofa  Population/solution.clj
(ns arge.core)

(defn nb-year [p0 percent aug p]
  ((fn [p0 percent aug p years]
      (if (> p p0)
        (recur (Math/floor (+ p0 aug (* p0 (/ percent 100)))) percent aug p (inc years))
        years)) p0 percent aug p 0))

File: 7KYU/Growth ofa  Population/solution.go
package kata

import "math"

func NbYear(p0 int, percent float64, aug int, p int) int {
	years := 0
	for p0 < p {
		p0 = int(math.Floor(float64(p0) + float64(aug) + (float64(p0) * (percent / 100.0))))
		years++
	}
	return years
}
